### ISSUE TITLE
Debug: execvp(3) failed.: No such file or directory

### ISSUE THREAD

**Issue by chrmarti** [Microsoft/vscode#63104]
I'm getting a very similar error in today's insider's build when trying to debug a C# .NET Core web app. The regular build of vscode is working fine. Not sure if I should open a new ticket, or if it's truly related to what @rogersachan is reporting here?

- VSCode Version: 1.30.0-insider (1.30.0-insider)
- OS Version: macOS Mojave 10.14

```
> Executing task: ${__process__} build /Users/tom/Documents/Repositories/code/Web/Web.csproj /property:GenerateFullPaths=true <

execvp(3) failed.: No such file or directory
The terminal process terminated with exit code: 1
```

My "terminal.integrated.shell.linux" setting is "/bin/bash".

_Originally posted by @winzig in https://github.com/Microsoft/vscode/issues/62699#issuecomment-438360045_

**Comment by weinand** [comment]
This seems to be a problem with tasks: If I run the "Build" action in C# project, I see this error:

![2018-11-14_10-02-02](https://user-images.githubusercontent.com/1898161/48471540-65c13f00-e7f4-11e8-8c47-8cfec695ef4c.png)

The problem is independent from debugging, but if the prelaunch task of a debug config uses the build task, the problem shows up there.

**Comment by weinand** [comment]
This was my test project:
[myApp.zip](https://github.com/Microsoft/vscode/files/2580021/myApp.zip)

Please note that I'm on macOS...

**Comment by alexr00** [comment]
This appears to be related to the newly added 

    private static ProcessVarName = '${__process__}';

It appears not to have any handling on non-Windows. I'm still investigating. FYI @dbaeumer

**Comment by sikhote** [comment]
Seeing this issue in Ubuntu 18 with bash or zsh used:
```
Version: 1.29.1
Commit: bc24f98b5f70467bc689abf41cc5550ca637088e
Date: 2018-11-18T13:10:54.096Z
Electron: 2.0.12
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64
```

**Comment by alexr00** [comment]
@sikhote, this is fixed in the Insiders build which you can download here if you want to try it out: https://code.visualstudio.com/insiders/
It will be fixed in our next release too.

**Comment by sikhote** [comment]
FWIW I was having this issue in 1.30 as well when trying to use `/usr/bin/zsh`. It seems the editor itself did not have access to all the files in `/bin`, so it could not find `zsh`. This also explains why when using `/bin/bash` I was not able to load up some paths from `/bin`.

I had installed VSCode through Flatpak, so that seems to be the reason.

I just installed through the `.deb` file and every problem went away.
